source/melpomene/src/main.rs

```diff
@@ -324,6 +325,11 @@ async fn kernel_entry(opts: MelpomeneOptions) {
         // If there is nothing else scheduled, and we didn't just wake something up,
         // sleep for some amount of time
         if turn.expired == 0 && !tick.has_remaining {
+            if !last_also_done {
```

Comment:
This isn't *really* a solution, it's just interesting that *one more poll* seems to be enough to prevent hangs, which could maybe help in diagnosing (it means that the "next)deadline" isn't calculated totally right, and/or maybe "expired" isn't reporting the case where it DID just wake something up, and we should go poll it?